server/kv.go

```diff
@@ -483,25 +463,18 @@ func (store *store) LoadInstance(instanceID types.ID) (Instance, error) {
 }
 
 func (store *store) LoadInstanceFullKey(fullkey string) (Instance, error) {
-	data, appErr := store.plugin.API.KVGet(fullkey)
-	if appErr != nil {
-		return nil, appErr
-	}
-	if data == nil {
-		return nil, errors.Wrap(kvstore.ErrNotFound, fullkey)
-	}
-
-	// Unmarshal into any of the types just so that we can get the common data
 	si := serverInstance{}
-	err := json.Unmarshal(data, &si)
+	err := store.plugin.client.KV.Get(fullkey, &si)
 	if err != nil {
 		return nil, err
 	}
-
+	if si.InstanceCommon == nil {
+		return nil, errors.Wrap(kvstore.ErrNotFound, fullkey)
+	}
 	switch si.Type {
 	case CloudInstanceType:
 		ci := cloudInstance{}
-		err = json.Unmarshal(data, &ci)
+		err := store.plugin.client.KV.Get(fullkey, &ci)
```

Comment:
Given that this is a kind of custom JSON unmarshal case, I would also be fine with simply getting a byte slice using `store.plugin.client.KV.Get` and keeping the rest of the code as it is.

Reply:
Thanks @hanzei. Reverted to use existing custom JSON unmarshal logic while using `store.plugin.client.KV.Get`